c-dependencies/js-compute-runtime/builtins/worker-location.cpp

```diff
@@ -0,0 +1,75 @@
+#include "builtins/worker-location.h"
+#include "builtin.h"
+#include "builtins/url.h"
+#include "host_call.h"
+#include "js-compute-builtins.h"
+
+/**
+ * The `WorkerLocation` builtin, added to the global object as the data property
+ * `location`.
+ * https://html.spec.whatwg.org/multipage/workers.html#worker-locations
+ */
+namespace builtins {
+JS::PersistentRooted<JSObject *> WorkerLocation::url;
+
+#define WorkerLocation_ACCESSOR_GET(field)                                                         \
+  bool field##_get(JSContext *cx, unsigned argc, JS::Value *vp) {                                  \
+    auto result = WorkerLocation::MethodHeaderWithName(0, cx, argc, vp, __func__);                 \
+    if (result.isErr()) {                                                                          \
+      return false;                                                                                \
+    }                                                                                              \
+    auto [args, self] = result.unwrap();                                                           \
+    REQUEST_HANDLER_ONLY("location." #field)                                                       \
+    return URL::field(cx, WorkerLocation::url, args.rval());                                       \
+  }
+
+WorkerLocation_ACCESSOR_GET(href) WorkerLocation_ACCESSOR_GET(origin)
+    WorkerLocation_ACCESSOR_GET(protocol) WorkerLocation_ACCESSOR_GET(host)
+        WorkerLocation_ACCESSOR_GET(hostname) WorkerLocation_ACCESSOR_GET(port)
+            WorkerLocation_ACCESSOR_GET(pathname) WorkerLocation_ACCESSOR_GET(search)
+                WorkerLocation_ACCESSOR_GET(hash)
```

Comment:
This is some pretty silly formatting from clang-format. Does it still work if you put a `;` on the end of each one?